src/internal/types.test.ts

```diff
@@ -156,3 +161,144 @@ describe("IfBoundedRecord", () => {
+    >().toEqualTypeOf<string>();
+
+    expectTypeOf<
+      EnumerableStringKeyOf<{ [SymbolFoo]: number; a: unknown }>
+    >().toEqualTypeOf<"a">();
+
+    expectTypeOf<
+      EnumerableStringKeyOf<Record<string | typeof SymbolFoo, unknown>>
+    >().toEqualTypeOf<string>();
+  });
+
+  test("optional keys", () => {
+    expectTypeOf<
+      EnumerableStringKeyOf<{ a: unknown; b?: unknown }>
+    >().toEqualTypeOf<"a" | "b">();
+  });
+
+  test("branded types", () => {
+    expectTypeOf<
+      EnumerableStringKeyOf<Record<Branded<string, symbol>, unknown>>
+    >().toEqualTypeOf<`${Branded<string, symbol>}`>();
+  });
+});
+
+describe("EnumerableStringKeyedValueOf", () => {
+  test("string values", () => {
+    expectTypeOf<
+      EnumerableStringKeyedValueOf<Record<number, string>>
+    >().toEqualTypeOf<string>();
+  });
+
+  test("number values", () => {
+    expectTypeOf<
+      EnumerableStringKeyedValueOf<Record<string, number>>
+    >().toEqualTypeOf<number>();
+  });
+
+  test("union of records", () => {
+    expectTypeOf<
+      EnumerableStringKeyedValueOf<
+        Record<number, string> | Record<string, string>
+      >
+    >().toEqualTypeOf<string>();
+
+    expectTypeOf<
+      EnumerableStringKeyedValueOf<
+        Record<number, string> | Record<string, number>
+      >
+    >().toEqualTypeOf<number | string>();
+  });
+
+  test("union values", () => {
+    expectTypeOf<
+      EnumerableStringKeyedValueOf<Record<number, number | string>>
+    >().toEqualTypeOf<number | string>();
+  });
+
+  test("literal values", () => {
+    expectTypeOf<EnumerableStringKeyedValueOf<{ a: 1 }>>().toEqualTypeOf<1>();
+
+    expectTypeOf<
+      EnumerableStringKeyedValueOf<{ a: "1" | "2" | 1 }>
+    >().toEqualTypeOf<"1" | "2" | 1>();
+  });
+
+  test("optional values", () => {
+    expectTypeOf<EnumerableStringKeyedValueOf<{ a: 1; b?: 4 }>>().toEqualTypeOf<
+      1 | 4
+    >();
+
+    expectTypeOf<
+      EnumerableStringKeyedValueOf<{ a: string; b?: number }>
+    >().toEqualTypeOf<number | string>();
+  });
+
+  test("nullish and undefined values", () => {
+    expectTypeOf<
+      EnumerableStringKeyedValueOf<{
+        a: string | undefined;
+        b: string | null;
+      }>
+    >().toEqualTypeOf<string | null | undefined>();
+
+    expectTypeOf<
+      EnumerableStringKeyedValueOf<{
+        a?: number | null;
+        b?: number | null | undefined;
+      }>
+    >().toEqualTypeOf<number | null | undefined>();
+  });
+
+  test("symbol keys", () => {
+    expectTypeOf<
+      EnumerableStringKeyedValueOf<{ [SymbolFoo]: string }>
+    >().toEqualTypeOf<never>();
+
+    expectTypeOf<
+      EnumerableStringKeyedValueOf<{ [SymbolFoo]: string; b: "1" }>
+    >().toEqualTypeOf<"1">();
+
+    expectTypeOf<
+      EnumerableStringKeyedValueOf<Record<string | typeof SymbolFoo, string>>
+    >().toEqualTypeOf<string>();
+  });
+
+  test("empty object", () => {
+    const emptyObj = {} as const;
```

Comment:
Doesn't type-fest already provide a type for this?

Reply:
Are you referring to the `EmptyObject` type? Sure, I could use that one, but it is also used in the `ValuesOf` type. So I thought I might utilize it this way, as I've seen it used in other tests. I could replace it with the `EmptyObject`.